backend/src/test/java/fk/prof/backend/ProfileApiTest.java

```diff
@@ -0,0 +1,453 @@
+import fk.prof.backend.service.IProfileWorkService;
+import fk.prof.backend.service.ProfileWorkService;
+import fk.prof.aggregation.cpusampling.CpuSamplingTraceDetail;
+import fk.prof.aggregation.cpusampling.CpuSamplingFrameNode;
+import io.vertx.core.*;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.junit.*;
+import org.junit.runner.RunWith;
+import recording.Recorder;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.zip.Adler32;
+import java.util.zip.Checksum;
+
+@RunWith(VertxUnitRunner.class)
+public class ProfileApiTest {
+
+  private static Vertx vertx;
+  private static Integer port;
+  private static IProfileWorkService profileWorkService;
+  private static AtomicLong workIdCounter = new AtomicLong(0);
+
+  @BeforeClass
+  public static void setUp(TestContext context) throws IOException {
+    ConfigManager.setDefaultSystemProperties();
+    JsonObject config = ConfigManager.loadFileAsJson(ProfileApiTest.class.getClassLoader().getResource("config.json").getFile());
+    JsonObject vertxConfig = config.getJsonObject("vertxOptions");
+    JsonObject deploymentConfig = config.getJsonObject("deploymentOptions");
+    assert deploymentConfig != null;
+
+    vertx = vertxConfig != null ? Vertx.vertx(new VertxOptions(vertxConfig)) : Vertx.vertx();
+    profileWorkService = new ProfileWorkService();
+    port = deploymentConfig.getJsonObject("config").getInteger("http.port");
+    DeploymentOptions deploymentOptions = new DeploymentOptions(deploymentConfig);
+    VertxManager.deployHttpVerticles(vertx, deploymentOptions, 2, profileWorkService);
+  }
+
+  @AfterClass
+  public static void tearDown(TestContext context) {
+    VertxManager.close(vertx);
+  }
+
+  @Test
+  public void testWithValidSingleProfile(TestContext context) {
+    long workId = workIdCounter.incrementAndGet();
+    profileWorkService.associateAggregationWindow(workId,
+        new AggregationWindow("a", "c", "p", LocalDateTime.now(), 20, 60, new long[]{workId}));
+
+    final Async async = context.async();
+    Future<Buffer> future = makeProfileRequest(context, MockProfileObjects.getRecordingHeader(workId), getMockWseEntriesForSingleProfile());
+    future.setHandler(ar -> {
+      if (ar.failed()) {
+        context.fail(ar.cause());
+      } else {
+        //Validate aggregation
+        AggregationWindow aggregationWindow = profileWorkService.getAssociatedAggregationWindow(workId);
+        validateAggregationBucketOfPredefinedSamples(context, aggregationWindow);
+
+        ProfileWorkInfo wi = aggregationWindow.getWorkInfo(workId);
+        validateWorkInfo(context, wi, 3);
+
+        async.complete();
```

Comment:
If this line isn't executed, this test will wait forever. We should provide a sensible timeout for it.